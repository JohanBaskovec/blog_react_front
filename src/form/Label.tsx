import React from "react";
import "./Label.scss";

interface LabelProps {
    style?: React.CSSProperties;
    children?: React.ReactNode;
    inputName: string;
}

export function Label(props: LabelProps): JSX.Element {
    return (<label className="Label" style={props.style} htmlFor={props.inputName}>
        {props.children}
    </label>)
}
