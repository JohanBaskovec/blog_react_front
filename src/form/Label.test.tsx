import React from "react";
import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect'
import {Label} from "./Label";

describe('Label', () => {
    it("is rendered properly", () => {
        const onClick = () => {
        }
        const tree = TestRenderer
            .create(<Label inputName="inputName">Hello</Label>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
