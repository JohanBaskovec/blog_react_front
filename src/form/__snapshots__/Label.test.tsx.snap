// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label is rendered properly 1`] = `
<label
  className="Label"
  htmlFor="inputName"
>
  Hello
</label>
`;
